Composer/packages/client/src/recoilModel/dispatchers/skill.ts

```diff
@@ -84,29 +63,37 @@ export const skillDispatcher = () => {
   );
 
   const removeSkill = useRecoilCallback(
-    (callbackHelpers: CallbackInterface) => async (projectId: string, manifestUrl?: string) => {
+    (callbackHelpers: CallbackInterface) => async (projectId: string, key: string) => {
       const { snapshot } = callbackHelpers;
-      const currentSkills = await snapshot.getPromise(skillsState(projectId));
-      const skills = currentSkills.filter((skill) => skill.manifestUrl !== manifestUrl);
-      await updateSkillState(callbackHelpers, projectId, skills);
+      const settings = await snapshot.getPromise(settingsState(projectId));
+      const newSettings = cloneDeep(settings);
+      delete newSettings.skill?.[key];
+
+      setSettingState(callbackHelpers, projectId, newSettings);
     }
   );
 
   const updateSkill = useRecoilCallback(
     (callbackHelpers: CallbackInterface) => async (
       projectId: string,
-      { targetId, skillData }: { targetId: number; skillData?: any }
+      key: string,
+      { endpointUrl, manifestUrl, msAppId, name }: SkillSetting
     ) => {
       const { snapshot } = callbackHelpers;
-      const originSkills = [...(await snapshot.getPromise(skillsState(projectId)))];
-
-      if (targetId >= 0 && targetId < originSkills.length && skillData) {
-        originSkills.splice(targetId, 1, skillData);
-      } else {
-        throw new Error(`update out of range, skill not found`);
-      }
-
-      updateSkillState(callbackHelpers, projectId, originSkills);
+      const settings = await snapshot.getPromise(settingsState(projectId));
+      const newSettings = cloneDeep(settings);
+      setSettingState(callbackHelpers, projectId, {
+        ...newSettings,
+        skill: {
+          ...newSettings?.skill,
+          [key]: {
```

Comment:
nitpick

We have been updating the specific property in the state like this in all dispatchers. But, we actually have immer. We can use the produce function and this section of the code would be more readable without using the spread operators.